src/components/TimePicker/Time/Time.tsx

```diff
@@ -0,0 +1,145 @@
+import { Component, Prop, Event } from '@/core';
+import { TimeAction } from './TimeAction';
+import { TimeInput } from './TimeInput';
+import { Watch } from 'vue-property-decorator';
+import { DateRangeMixin } from '../mixins/DateRange';
+import { mixins } from 'vue-class-component';
+
+interface Props {
+  type?: TimeType;
+  value?: string | number | null;
+  ariaLabel?: string | null;
+}
+
+// Time type
+const typeMapping = {
+  hour24: '24 Hour range',
+  hour12: '12 Hour range',
+  minute: 'Minute range',
+  second: 'Second range',
+  meridian: 'am/pm meridian range',
+};
+export type TimeType = keyof (typeof typeMapping);
+export const TimeTypeList = Object.keys(typeMapping) as TimeType[];
+
+const TimeBase = mixins(DateRangeMixin);
+
+@Component('Time')
+@Event('timeUpdate', 'Event triggered whenever the time value is updated.')
+export class Time extends TimeBase<Props> {
+  @Prop('Time Item Type', {
+    type: String,
+    required: true,
+    acceptableValues: TimeTypeList,
+    validator: TimeTypeList.includes,
+  })
+  public type!: TimeType;
+
+  @Prop('Value in the Time input field', {
+    type: [String, Number],
+    default: '--',
+  })
+  public value!: string | number;
+
+  @Prop('Aria Label', {
+    type: String,
+    default: 'Time Item',
+  })
+  public ariaLabel!: string | null;
+
+  public $tsxProps!: Readonly<{}> & Readonly<Props>;
+  private inputValue: string | number = this.value;
+
+  private getPreviousValue() {
+    const previousValue = Number(this.inputValue) > Number(this.getRange()[this.type].min) ? Number(this.inputValue) - 1 : this.getRange()[this.type].max;
+    return previousValue.toString().padStart(2, '0');
+  }
+
+  private getNextValue() {
```

Comment:
I think all those `getXXX`-methods should actually be simple getters. At least according to the vue-class-component documentation. https://github.com/vuejs/vue-class-component